### ISSUE TITLE
Sync does not seem to respect source configuration config.toml

### ISSUE THREAD

**Issue by moonkev** [intellij-rust/intellij-rust#8136]
## Environment

* **IntelliJ Rust plugin version:** 0.4.159.4232-212
* **Rust toolchain version:** 0.56.1
* **IDE name and version:** Intellij IDEA 2021.2 (Community Edition)
* **Operating system:** Windows 10

## Problem description

When running the Sync task, it seems that Cargo is not picking up [source] entries config.toml (regardless if it is in ./.cargo/config.toml relative to the project or ~/.cargo/config.toml).   Due to being behind a restrictive corporate proxy at work, we are not able to clone repositories from github over http.  To get around this, we are hosting it on an internal git server, and are overriding the registry for crates-io like so

```toml
[source]

[source.internal-crates-io]
registry = "ssh://git@internal.repo/crates-io.index.git"

[source.crates-io]
replace-with = "local-crates-io"
```

This configuration gets picked up if we build/run, it only seems to affect the sync task.

## Steps to reproduce

Override crates-io registry as detailed above.

**Comment by moonkev** [comment]
As a work around, in case anyone else is facing the same issue, I was able to use insteadOf in my ~/.gitcconfig

```
[url ="ssh://git@internal.repo/crates-io.index.git"]
    insteadOf = https://github.com/rust-lang/crates-io.index
```

**Comment by Undin** [comment]
Fixes in #8578
